relay-server/src/actors/processor.rs

```diff
@@ -2750,22 +2764,38 @@ impl EnvelopeProcessorService {
 impl Service for EnvelopeProcessorService {
     type Interface = EnvelopeProcessor;
 
-    fn spawn_handler(self, mut rx: relay_system::Receiver<Self::Interface>) {
+    fn spawn_handler(mut self, mut rx: relay_system::Receiver<Self::Interface>) {
         let thread_count = self.inner.config.cpu_concurrency();
         relay_log::info!("starting {thread_count} envelope processing workers");
 
         tokio::spawn(async move {
             let semaphore = Arc::new(Semaphore::new(thread_count));
 
-            while let (Some(message), Ok(permit)) =
-                tokio::join!(rx.recv(), semaphore.clone().acquire_owned())
-            {
-                let service = self.clone();
+            let mut global_config_rx = self
```

Comment:
In addition to retrieving the rx, you should await it once to ensure you have an up-to-date config. That is, unless you want to process with default config, but that can be dangerous. 

Of course, the GlobalConfigService should still be able to send you a default config under certain conditions (e.g. in proxy mode), but that's not a concern of the EnvelopeProcessor.